core-common/src/main/java/org/zalando/nakadi/annotations/validation/DataLakeAnnotationValidator.java

```diff
@@ -7,29 +7,47 @@
 
 public class DataLakeAnnotationValidator implements ConstraintValidator<DataLakeValidAnnotations, Map<String, String>> {
     private static final Pattern ANNOTATIONS_PERIOD_PATTERN = Pattern.compile(
-            "^(unlimited|(([1-9]\\d{0,2}|[1-2]\\d{3}|3[0-5]\\d{2}|36[0-4]\\d|3650)(\\sdays?))|" +
-                    "(([1-9]|[1-9]\\d|[1][01]\\d|120)(\\smonths?))|(([1-9]|(10))(\\syears?)))$");
+            "^(unlimited|(([7-9]|[1-9]\\d{1,2}|[1-2]\\d{3}|3[0-5]\\d{2}|36[0-4]\\d|3650)((\\sdays?)|(d)))" +
+                    "|(([1-9][0-9]?|[1-4][0-9]{2}|5([0-1][0-9]|2[0-1]))((\\sweeks?)|(w)))|" +
+                    "(([1-9]|[1-9]\\d|[1][01]\\d|120)((\\smonths?)|(m)))|(([1-9]|(10))((\\syears?)|(y))))$");
     public static final String RETENTION_PERIOD_ANNOTATION = "datalake.zalando.org/retention-period";
     public static final String RETENTION_REASON_ANNOTATION = "datalake.zalando.org/retention-period-reason";
+    public static final String MATERIALISE_EVENTS_ANNOTATION = "datalake.zalando.org/materialize-events";
 
     @Override
     public boolean isValid(final Map<String, String> annotations, final ConstraintValidatorContext context) {
         if (annotations == null || annotations.size() == 0) {
             return true;
         }
+        if (annotations.containsKey(MATERIALISE_EVENTS_ANNOTATION)) {
+            if (!annotations.get(MATERIALISE_EVENTS_ANNOTATION).equals("off") &&
+                    !annotations.get(MATERIALISE_EVENTS_ANNOTATION).equals("on")) {
+                context.disableDefaultConstraintViolation();
+                context.buildConstraintViolationWithTemplate("Field " + MATERIALISE_EVENTS_ANNOTATION
+                                + " is not valid. Provided value: \""
+                                + annotations.get(MATERIALISE_EVENTS_ANNOTATION)
+                                + "\". Possible values are: \"on\" or \"off\".")
```

Comment:
Minor: I think mentioning `Field` might mislead (error) readers.
Suggestion to use `Annotation` here.

Same for the remaining constraint violation messages.